plugins/inputs/tail/tail_test.go

```diff
@@ -800,67 +800,67 @@ func TestStatePersistence(t *testing.T) {
-		file              string
-		InitialReadOffset string
-		expected          *tail.SeekInfo
+		name     string
+		offsets  map[string]int64
+		file     string
+		initial  string
+		expected *tail.SeekInfo
 	}{
 		{
-			name:              "Read from beginning when initial_read_offset set to beginning",
-			offsets:           map[string]int64{"test.log": 100},
-			file:              "test.log",
-			InitialReadOffset: "beginning",
+			name:    "Read from beginning when initial_read_offset set to beginning",
+			offsets: map[string]int64{"test.log": 100},
+			file:    "test.log",
+			initial: "beginning",
 			expected: &tail.SeekInfo{
 				Whence: 0,
 				Offset: 0,
 			},
 		},
 		{
-			name:              "Read from end when initial_read_offset set to end",
-			offsets:           map[string]int64{"test.log": 100},
-			file:              "test.log",
-			InitialReadOffset: "end",
+			name:    "Read from end when initial_read_offset set to end",
+			offsets: map[string]int64{"test.log": 100},
+			file:    "test.log",
+			initial: "end",
 			expected: &tail.SeekInfo{
 				Whence: 2,
 				Offset: 0,
 			},
 		},
 		{
-			name:              "Read from end when offset not exists and initial_read_offset set to saved-or-end",
-			offsets:           map[string]int64{},
-			file:              "test.log",
-			InitialReadOffset: "saved-or-end",
+			name:    "Read from end when offset not exists and initial_read_offset set to saved-or-end",
+			offsets: map[string]int64{},
+			file:    "test.log",
+			initial: "saved-or-end",
 			expected: &tail.SeekInfo{
 				Whence: 2,
 				Offset: 0,
 			},
 		},
 		{
-			name:              "Read from offset when offset exists and initial_read_offset set to saved-or-end",
-			offsets:           map[string]int64{"test.log": 100},
-			file:              "test.log",
-			InitialReadOffset: "saved-or-end",
+			name:    "Read from offset when offset exists and initial_read_offset set to saved-or-end",
+			offsets: map[string]int64{"test.log": 100},
+			file:    "test.log",
+			initial: "saved-or-end",
 			expected: &tail.SeekInfo{
 				Whence: 0,
 				Offset: 100,
 			},
 		},
 		{
-			name:              "Read from start when offset not exists and initial_read_offset set to save-offset-or-start",
-			offsets:           map[string]int64{},
-			file:              "test.log",
-			InitialReadOffset: "saved-or-beginning",
+			name:    "Read from start when offset not exists and initial_read_offset set to save-offset-or-start",
+			offsets: map[string]int64{},
+			file:    "test.log",
+			initial: "saved-or-beginning",
 			expected: &tail.SeekInfo{
 				Whence: 0,
 				Offset: 0,
 			},
 		},
 		{
-			name:              "Read from offset when offset exists and initial_read_offset set to saved-or-end",
-			offsets:           map[string]int64{"test.log": 100},
-			file:              "test.log",
-			InitialReadOffset: "saved-or-beginning",
+			name:    "Read from offset when offset exists and initial_read_offset set to saved-or-end",
```

Comment:
test name mentions "saved-or-end" but the code uses "saved-or-beginning"

Reply:
Fixed.